selftests/all/unit/avocado/tree_unittest.py

```diff
@@ -165,6 +166,21 @@ def test_advanced_yaml(self):
                          oldroot.children[1].children[2].value)
         self.assertEqual({'new_value': 'something'},
                          oldroot.children[3].children[0].children[0].value)
+        # multiplex root (always True)
+        self.assertEqual(tree2.multiplex, True)
+        # multiplex /virt/
+        self.assertEqual(tree2.children[0].multiplex, True)
+        # multiplex /virt/hw
+        self.assertEqual(tree2.children[0].children[0].multiplex, True)
+        # multiplex /virt/distro
+        self.assertEqual(tree2.children[0].children[1].multiplex, False)
+        # multiplex /virt/env
+        self.assertEqual(tree2.children[0].children[2].multiplex, False)
+        # multiplex /virt/absolutly
+        self.assertEqual(tree2.children[0].children[3].multiplex, True)
+        # multiplex /virt/distro/fedora
+        self.assertEqual(tree2.children[0].children[1].children[0].multiplex,
+                         True)
 
 
 class TestPathParent(unittest.TestCase):
```

Comment:
Well it's not separate project so one shouldn't assume it's generic. The tree parser was never really independent. It was tuned to our purposes with all the environment. Currently it supports 5 custom tags, one of them is `multiplex`.

So is there some general rule about independent and avocado-related code? If so I can move `tree.py` somewhere else. It's not completely dependant on multiplex, it just supports some custom tags. Some of them are related to `avocado`, some only to `multiplexer`. In the future I'd like to also separate filters from environment to custom tag to get rid of all reserved keys and use tags instead.